packages/ng-excel-builder/src/excel-builder.service.ts

```diff
@@ -155,27 +160,40 @@ export class ExcelBuilderService {
                     }
 
                     private getWidthOfColumn(
-                        key: string,
-                        entries: PlainObject[],
+                        title: string,
+                        entries: string[],
                         parameters: Nullable<ColumnParameters>
                     ): number {
                         const { minColumnWidth }: StyleSizes = this.sizes;
 
                         if (parameters?.width === ColumnWidth.MAX_WIDTH) {
-                            return this.calcMaxWidthByEntries(entries, key);
+                            return this.calcMaxWidthByEntries(entries, title);
                         } else if (typeof parameters?.width === 'number') {
                             return parameters.width;
                         } else {
                             return minColumnWidth;
                         }
                     }
 
-                    private calcMaxWidthByEntries(entries: PlainObject[], key: string): number {
-                        const maxLength: number = entries.reduce((length: number, entry: PlainObject): number => {
-                            const currentLength: number = (entry[key]?.toString() ?? '').length;
+                    private calcMaxWidthByEntries(entries: string[], title: string): number {
+                        const titleLength: number = this.getWidthOfString(title, 'bold');
+                        // eslint-disable-next-line @typescript-eslint/no-magic-numbers
+                        const padding: number = this.sizes.fontWidth * 2;
+                        const maxLength: number = entries.reduce((length: number, entry: string): number => {
+                            const currentLength: number = this.getWidthOfString(entry, 'regular');
                             return Math.max(currentLength, length);
-                        }, 0);
-                        return maxLength * this.sizes.fontWidth;
+                        }, titleLength);
+                        return Math.round(maxLength) + padding;
+                    }
+
+                    private getWidthOfString(string: string, fontWeight: keyof WidthOfSymbols): number {
+                        return string
+                            .split('')
+                            .reduce(
+                                (sum: number, symbol: string): number =>
+                                    sum + (input.widthOfSymbols[fontWeight][symbol] ?? this.sizes.fontWidth),
+                                0
+                            );
```

Comment:
i think it's more readable 
```ts
function getWidthOfString(string: string, fontWeight: keyof WidthOfSymbols): number {
    let width: number = 0;
    for (const symbol of string) {
        width += input.widthOfSymbols[fontWeight][symbol] ?? this.sizes.fontWidth;
    }
    return width;
}
```

Reply:
`input.widthOfSymbols?.[fontWeight]?.[symbol]`

Reply:
@StepaZol ))

Reply:
I like reduce)